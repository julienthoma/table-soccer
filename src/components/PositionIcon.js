import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import PlayerIcon from './PlayerIcon';

const Container = styled.div`
  display: flex;
  justify-content: center;
`;

const PositionIcon = ({ count, dark }) => (
  <Container>
    {[...Array(count)].map((e, i) => (
      // eslint-disable-next-line
      <PlayerIcon key={i} dark={dark} />
    ))}
  </Container>
);

PositionIcon.defaultProps = {
  dark: false,
  count: 1
};

PositionIcon.propTypes = {
  count: PropTypes.number,
  dark: PropTypes.bool
};

export default PositionIcon;
