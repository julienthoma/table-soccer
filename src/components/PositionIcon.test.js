import React from 'react';
import { renderWithMuiTheme } from '../test.utils';
import PositionIcon from './PositionIcon';

it('PositionIcon renders correctly', () => {
  expect(renderWithMuiTheme(<PositionIcon count={3} dark />)).toMatchSnapshot();
});
