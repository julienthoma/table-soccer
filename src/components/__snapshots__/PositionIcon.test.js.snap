// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionIcon renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1px;
}

.c2 {
  background: #c5c5c5;
  height: 8px;
  width: 8px;
  border-radius: 50px;
}

.c3 {
  background: #c5c5c5;
  margin-top: 1px;
  width: 10.4px;
  height: 14.4px;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    />
  </div>
</div>
`;
